src/r1cs.rs

```diff
@@ -331,6 +344,80 @@ pub fn r1cs_program<T: Field>(
     (variables, private_inputs_offset, a, b, c)
 }
 
+
+/// Calculates a flattend program based on a R1CS (A, B, C) and returns that flattened program:
+/// * The Rank 1 Constraint System (R1CS) is defined as:
+/// * `<A,x>*<B,x> = <C,x>` for a witness `x`
+/// * Since the matrices in R1CS are usually sparse, the following encoding is used:
+/// * For each constraint (i.e., row in the R1CS), only non-zero values are supplied and encoded as a tuple (index, value).
+///
+/// # Arguments
+///
+/// * r1cs - R1CS in standard JSON data format
+
+pub fn flattened_program<T: Field>(
```

Comment:
Could this be in an `impl` for `R1CS` to call `R1CS::from(json).to_flattened_program()` ?

Reply:
Yes, good idea. Let's add that after the format has been standardized and the data structure fixed.

Reply:
You mean not in this PR?

Reply:
That depends if we want to merge this preliminary version or finalize it first.